### DISCUSSION TITLE
Convert into a python project

### DISCUSSION THREAD

**Discussion by vitoyucepi** [Moonbase59/autocue#23] [General]
## Summary
Does it make sense to convert this repository into the python project with poetry and `pyproject.toml`?

### Pros:
- Allows easy dependency management.
- Standardized way to build packages.
- Makes it easier to use advanced python features.
### Cons:
- Increases the entry level.
- Tools maintenance.

I have limited experience in python projects and insufficient understanding of github actions (I use gitlab), but I would like to create some PRs that would split the code, bring various tools and implement basic ci pipeline.

**Comment by Moonbase59** [comment]
Pandora’s Box… Bear with me, I’m an old man of severely bad health, and I’m already investing too much lifetime into this, much more than is good for me, having to struggle to survive month after month. And I’m currently rather burned out.

That said, I much appreciate your input, and it might be a good idea to put it on PyPi or the like, but I have not even an idea how to work with PRs correctly, despite using all these modern build tools. I also rather like just having _one_ small program that does _one_ thing in the best way possible.

Then again, as you say, things would split and be possible harder to maintain. I mean, some Python here, some Liquidsoap there—is that good?

I don’t want to turn down the idea, I just don’t feel up to it right now. Leaving this open for further discussion/talk/whatever.

**Comment by vitoyucepi** [comment]
> Then again, as you say, things would split and be possible harder to maintain. I mean, some Python here, some Liquidsoap there—is that good?

At first I thought that creating two separate repositories to track the different parts of the project separately would be a good idea, but then I realized that there's no need for that; it's possible to have liquidsoap scripts somewhere in the subdirectory.

And of course, it will require not only python and a simple text editor to maintain, but also a bunch of tools and utilities. One of them would be poetry, which plays the role of npm in the python world.

> I also rather like just having one small program that does one thing in the best way possible.

That's fine, but I still don't understand the point of having two main branches.

**Reply by Moonbase59** [reply]
> That's fine, but I still don't understand the point of having two main branches.

That was just something that evolved gradually while we were developing the Liquidsoap integration, and is now fixed: Master is master again. ;-)
